libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -170,6 +170,20 @@ float SRV_Channel::get_output_norm(void)
     return ret;
 }
 
+float SRV_Channel::get_output_norm_from_pwm(uint16_t pwm)
+{
+    float ret;
+    if (pwm < servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_trim - servo_min);
+    } else if (pwm > servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_max  - servo_trim);
+    } else {
+        ret = 0;
+    }
+    
```

Comment:
1) Regarding trim as being middle, this is common:
![image](https://cloud.githubusercontent.com/assets/7996972/24745591/e246e43a-1abe-11e7-93d0-8f3f5c0a2165.png)
The "trim" itself is actually the middle, neutral position.
I would like to hear @tridge about this, as on my planes, setting same angle of ailerons and other surfaces, have different angle of servo [(max-trim) != (trim-min)] to get same angle of deflection.

2) I have a situation where I will have both servos and ESCs so I disagree with:
> all boards that need ESC scaling only have ESCs and no servos
I think it is more correct to say that we will not have both servo and ESC on same channel at same time.
Also in HAL you have 1 set of min/max for all ESCs. I think this is not correct either. For example, let's take quad plane. 4 ESCs on wings and 1 (or 2) ESCs for horizontal thrust. They might have different settings.
The reason I did this way, was to have separate control for each channel of servo or ESC. I still insist that this approach is much more universal and I think one set of settings for all ESCs is legacy.

3) Constrain ret value to -1/1 will be added.

4) As for reverse: this method is called with a value taken from HAL::RCOutput. So this PWM value is actually the one that is already going out of the board and it **was** reversed already if it was needed.